src/adaXT/random_forest/random_forest.py

```diff
@@ -39,32 +38,52 @@ def tree_based_weights(
 def get_sample_indices(
     gen: Generator,
     n_rows: int,
-    sampling_parameter: int | tuple[int, int],
+    sampling_args: dict,
     sampling: str | None,
 ) -> Iterable:
     """
     Assumes there has been a previous call to self.__get_sample_indices on the
     RandomForest.
     """
-    if sampling == "bootstrap":
-        return (
-            gen.integers(
-                low=0,
-                high=n_rows,
-                size=sampling_parameter),
-            None)
+    if sampling == "resampling":
+        return (gen.choice(np.arange(0, n_rows),
+                           size=sampling_args['size'],
+                           replace=sampling_args['replace']), None)
     elif sampling == "honest_tree":
         indices = np.arange(0, n_rows)
         gen.shuffle(indices)
```

Comment:
no need to do the shuffle if you decide to now use gen.choice later to get the fit_indices, pred_indices. However, in the case that replace is false, it might be faster to do the shuffle still instead of doing the random choice.

Reply:
This is needed so that for each tree we draw from different sets of indices for fitting and prediction. Its the only difference between honest_tree and honest_forest.

Reply:
yes, but then in the case of doing sampling with "replace" there is no reason to do the random choice, no?
I mean the random shuffle would randomly shuffle all the indices, and then one could simply return first half as the fitting indices and the second half as prediction indices. It is only for when resampling is false, that we would need to still do the random choice again. With the current implementation we are adding another layer of randomness, which I don't see a reason for?

Reply:
Yes that is certainly true but adding an additional step might make the code more confusing (the difference between honest_tree and honest_forest becomes less clear then) and I don't think the computation is a big bottleneck here.